src/main/java/frc/robot/DriveBase.java

```diff
@@ -4,61 +4,76 @@
 import edu.wpi.first.wpilibj.CounterBase.EncodingType;
 import edu.wpi.first.wpilibj.Encoder;
 import edu.wpi.first.wpilibj.Solenoid;
-import edu.wpi.first.wpilibj.VictorSP;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
 
 public class DriveBase {
 
-    private final VictorSP driveLeft;
-    private final VictorSP driveRight;
-    private final Solenoid shifter;
+    private final SpeedControllerGroup driveLeft;
+    private final SpeedControllerGroup driveRight;
+
+    private final Solenoid leftShifter;
+    private final Solenoid rightShifter;
 
     private final Encoder leftEncoder;
     private final Encoder rightEncoder;
 
-    private final ADXRS450_Gyro gyro;
-
     private boolean usingLeftEncoder = false;
 
     private double heading;
     public double courseHeading;
 
-    public DriveBase(int left, int right, int leftA, int leftB, int rightA, int rightB, int shifterChannel) {
-        driveLeft = new VictorSP(left);
-        driveRight = new VictorSP(right);
+    /**
+     * 
+     * @param driveLeft           positive moves forward. the front of the robot is
+     *                            opposite the intake.
+     * @param driveRight
+     * @param leftShifterChannel
+     * @param rightShifterChannel
+     * @param leftA
+     * @param leftB
+     * @param rightA
+     * @param rightB
+     */
+    public DriveBase(SpeedControllerGroup driveLeft, SpeedControllerGroup driveRight, int leftShifterChannel,
+            int rightShifterChannel, int leftA, int leftB, int rightA, int rightB) {
+        this.driveLeft = driveLeft;
+        this.driveRight = driveRight;
+        this.driveRight.setInverted(true);
+
+        leftShifter = new Solenoid(leftShifterChannel);
+        rightShifter = new Solenoid(rightShifterChannel);
 
         leftEncoder = new Encoder(leftA, leftB, false, EncodingType.k4X);
         rightEncoder = new Encoder(rightA, rightB, true, EncodingType.k4X);
 
         leftEncoder.setDistancePerPulse((4 * Math.PI) / 1024.0);
         rightEncoder.setDistancePerPulse((4 * Math.PI) / 1024.0);
-
-        shifter = new Solenoid(shifterChannel);
-
-        gyro = new ADXRS450_Gyro();
-        gyro.calibrate();
-        heading = gyro.getAngle();
-        courseHeading = heading;
     }
 
     public void drive(double leftPower, double rightPower) {
         driveLeft.set(leftPower);
-        driveRight.set(-rightPower);
+        driveRight.set(rightPower);
     }
 
     public void drive(double power) {
         drive(power, power);
     }
 
     public void useHighGear(boolean highGear) {
-        shifter.set(highGear);
+        leftShifter.set(highGear);
+        rightShifter.set(highGear);
+    }
+
+    public void toggleHighGear() {
+        useHighGear(!leftShifter.get());
     }
 
     public void resetGyro() {
-        gyro.reset();
+        // TODO: NOPE
```

Comment:
I do not love this. We may want to 1) remove the gyro methods, 2) implement the Analog Devices IMU, or 3) leave this, but open an issue to implement the gyro.

Reply:
3